docs/WindowsBuild.md

```diff
@@ -0,0 +1,176 @@
+- Using the latest Visual Studio version is recommended. Swift may fail to
+  build with older C++ compilers.
+
+### 1. Install dependencies
+- Make sure to add Python, CMake and Ninja to your `Path` environment variable
+1. Latest version (2.7.12 tested) of [Python
+   2](https://www.python.org/downloads/)
+2. Latest version (3.7.0-rc3 tested) of [CMake](https://cmake.org/download/)
+3. Latest version (1.7.1 tested) of
+   [Ninja](https://github.com/ninja-build/ninja/releases/latest)
+4. Latest version (2015 Update 3 tested) of [Visual
+   Studio](https://www.visualstudio.com/downloads/)
+
+- Make sure to include `Programming Languages|Visual C++` and `Windows and Web
+  Development|Universal Windows App Development|Windows SDK` in your
+  installation.
+
+### 2. Clone the repositories
+1. Create a folder to contain all the Swift repositories
+2. `apple/swift-cmark` into a folder named `cmark`
+3. `apple/swift-clang` into a folder named `clang`
+5. `apple/swift-llvm` into a folder named `llvm`
+5. `apple/swift` into a folder named `swift`
+- Currently, other repositories in the Swift project have not been tested and
+  may not be supported.
+
+### 3. Build ICU
+1. Download and extract the [ICU source
+code](http://site.icu-project.org/download) to a folder named `icu` in the same
+directory as the other Swift project repositories.
+2. Open `src/win32/allinone.sln` in Visual Studio.
+3. Make sure to select the correct architecture from the drop-down in Visual
+Studio.
+4. Right click on the solution in the Solution Explorer window and select
+`Build Solution`.
+5. When this is done, add the `<icu-source>/bin` folder to your `Path`
+environment variable.
+
+### 4. Get ready
+- From within a **developer** command prompt (not PowerShell nor cmd, but [the
+  Visual Studio Developer Command
+  Prompt](https://msdn.microsoft.com/en-us/library/f35ctcxw.aspx)), execute the
+  following command if you have an x64 PC.
+```cmd
+VsDevCmd -arch=amd64
+```
+If instead, you're compiling for a 32-bit Windows target, adapt the `arch`
+argument to `x86` and run
+```cmd
+VsDevCmd -arch=x86
+```
+
+- Then adapt the following command and run it.
+```cmd
+set swift_source_dir=path-to-directory-containing-all-cloned-repositories
+```
+
+### 5. Build CMark
+- This must be done from within a developer command prompt. CMark is a fairly
+  small project and should only take a few minutes to build.
+```cmd
+mkdir "%swift_source_dir%/build/Ninja-DebugAssert/cmark-windows-amd64"
+pushd "%swift_source_dir%/build/Ninja-DebugAssert/cmark-windows-amd64"
+cmake -G "Ninja" "%swift_source_dir%/cmark"
+popd
+cmake --build "%swift_source_dir%/build/Ninja-DebugAssert/cmark-windows-amd64/"
+```
+
+### 6. Build LLVM/Clang/Compiler-RT
+- This must be done from within a developer command prompt. LLVM and Clang are
+  large projects so building might take a few hours. Make sure that the build
+  type (e.g. Debug/Release) for LLVM/Clang matches the build type for Swift.
```

Comment:
Do you mean `i.e.` instead of `e.g.`?

Reply:
No, I think Debug and Release are examples from a larger set, I'll add another example to make it more clear